CodeQL crash with Java 17

```
[2023-08-05 17:45:21] [autobuild] Caused by: org.codehaus.plexus.compiler.CompilerException: error: release version 17 not supported
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:198)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:183)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1140)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:193)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:299)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:193)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:106)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:963)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:296)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:199)
  [2023-08-05 17:45:21] [autobuild]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
  [2023-08-05 17:45:21] [autobuild]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
  [2023-08-05 17:45:21] [autobuild]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
  [2023-08-05 17:45:21] [autobuild]     at java.lang.reflect.Method.invoke (Method.java:566)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
  [2023-08-05 17:45:21] [autobuild] Caused by: java.lang.IllegalArgumentException: error: release version 17 not supported
  [2023-08-05 17:45:21] [autobuild]     at com.sun.tools.javac.main.Arguments.reportDiag (Arguments.java:891)
  [2023-08-05 17:45:21] [autobuild]     at com.sun.tools.javac.main.Arguments.handleReleaseOptions (Arguments.java:311)
  [2023-08-05 17:45:21] [autobuild]     at com.sun.tools.javac.main.Arguments.processArgs (Arguments.java:350)
  [2023-08-05 17:45:21] [autobuild]     at com.sun.tools.javac.main.Arguments.init (Arguments.java:246)
  [2023-08-05 17:45:21] [autobuild]     at com.sun.tools.javac.api.JavacTool.getTask (JavacTool.java:185)
  [2023-08-05 17:45:21] [autobuild]     at com.sun.tools.javac.api.JavacTool.getTask (JavacTool.java:119)
  [2023-08-05 17:45:21] [autobuild]     at com.sun.tools.javac.api.JavacTool.getTask (JavacTool.java:68)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:135)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:183)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1140)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:193)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:299)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:193)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:106)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:963)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:296)
  [2023-08-05 17:45:21] [autobuild]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:199)
  [2023-08-05 17:45:21] [autobuild]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
  [2023-08-05 17:45:21] [autobuild]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
  [2023-08-05 17:45:21] [autobuild]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
  [2023-08-05 17:45:21] [autobuild]     at java.lang.reflect.Method.invoke (Method.java:566)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
  [2023-08-05 17:45:21] [autobuild]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -26,6 +26,10 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v3
 
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 17
+
     # Initializes the CodeQL tools for scanning.
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v2

